conformance/src/main/scala/za/co/absa/enceladus/conformance/interpreter/DynamicInterpreter.scala

```diff
@@ -117,14 +153,21 @@ object DynamicInterpreter {
     }
   }
 
+  /** Applies a control framework checkpoint given a stage of the conformance */
+  private def applyCheckpoint(df: Dataset[Row], jobStage: String)(implicit ictx: InterpreterContext): Unit = {
+    if (ictx.enableControlFramework) {
+      df.setCheckpoint(s"${ictx.jobShortName} - $jobStage")
+    }
+  }
+
   /**
-    * applyCheckpoint Function which takes a generic rule and the dataframe and applies the control framework
+    * applyRuleCheckpoint Function which takes a generic rule and the dataframe and applies the control framework
     * checkpoint if configured
     *
     * @param rule The conformance rule
     * @param df   Dataframe to apply the checkpoint on
```

Comment:
Yeah, and also I can switch this function to take an implicit context as well, instead of passing 2 additional parameters (`encbleCF` and `shortJobName`). It makes sense considering how this function is used. These 2 parameters are being passed from a context.